test/undici-timeout.js

```diff
@@ -31,7 +31,7 @@ async function main () {
   instance.register(From, {
     base: `http://localhost:${target.server.address().port}`,
     undici: {
-      requestTimeout: 100
+      headersTimeout: 100
```

Comment:
could you add a copy of this test with `bodyTimeout`  set instead? Thanks

Reply:
I created a test server that sends the headers before the body as follows:

```js
const target = http.createServer((req, res) => {
  t.pass('request proxied')
  req.on('data', () => undefined)
  req.on('end', () => {
    res.flushHeaders();
    clock.setTimeout(() => {
      res.end(JSON.stringify({ something: 'else' }));
      t.end()
    }, 1000)
  })
})
```

however `await got.get(`http://localhost:${instance.server.address().port}/`, { retry: 0 })` now throws with 

> RequestError: Parse Error: Content-Length can't be present with Transfer-Encoding

Am I right in thinking that the headers are forwarded to the client as soon as they are received, and then when this runs:

https://github.com/penx/fastify-reply-from/blob/bf4793159d1864f36e366c366098e60eccee99bd/index.js#L179-L181

We get an error because the headers have already been sent?

Reply:
Can you include a full stacktrace?

Reply:
```
RequestError: Parse Error: Content-Length can't be present with Transfer-Encoding
    at ClientRequest.<anonymous> (/Users/penx/Development/forks/fastify-reply-from/node_modules/got/dist/source/core/index.js:953:111)
    at Object.onceWrapper (node:events:485:26)
    at ClientRequest.emit (node:events:390:22)
    at ClientRequest.origin.emit (/Users/penx/Development/forks/fastify-reply-from/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at Socket.socketOnData (node:_http_client:509:9)
    at Socket.emit (node:events:378:20)
    at addChunk (node:internal/streams/readable:313:12)
    at readableAddChunk (node:internal/streams/readable:288:9)
    at Socket.Readable.push (node:internal/streams/readable:227:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
    at Socket.socketOnData (node:_http_client:502:22)
    at Socket.emit (node:events:378:20)
    at addChunk (node:internal/streams/readable:313:12)
    at readableAddChunk (node:internal/streams/readable:288:9)
    at Socket.Readable.push (node:internal/streams/readable:227:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
    at TCP.callbackTrampoline (node:internal/async_hooks:132:14) {
  code: 'HPE_UNEXPECTED_CONTENT_LENGTH',
  timings: {
    start: 1612976101252,
    socket: 1612976101253,
    lookup: 1612976101254,
    connect: 1612976101255,
    secureConnect: undefined,
    upload: 1612976101255,
    response: undefined,
    end: undefined,
    error: 1612976103275,
    abort: undefined,
    phases: {
      wait: 1,
      dns: 1,
      tcp: 1,
      tls: undefined,
      request: 0,
      firstByte: undefined,
      download: undefined,
      total: 2023
    }
   }
 }
```

Reply:
I think you might need to remove `transfer-encoding` from `headers` in https://github.com/fastify/fastify-reply-from/blob/d4729809effa27d7429e41098938c0bd2dca4051/lib/request.js#L114.

Reply:
This helps, but then the `UND_ERR_BODY_TIMEOUT` can't be intercepted without an unwanted performance hit, as far as I can tell.

It seems to land here:
https://github.com/penx/fastify-reply-from/blob/bf4793159d1864f36e366c366098e60eccee99bd/index.js#L107-L107

with `err` being null, and then here:

https://github.com/penx/fastify-reply-from/blob/bf4793159d1864f36e366c366098e60eccee99bd/index.js#L136-L136

so starts streaming whatever it receives as the body from undici, in this case:

```json
{
   "statusCode": 500,                
   "code": "UND_ERR_BODY_TIMEOUT",   
   "error": "Internal Server Error", 
   "message": "Body Timeout Error",  
} 
```

The only approach to intercept this I guess would be to wait for the stream to complete first. Pretty sure this wouldn't be wanted so I guess we have the unit test expect this response?

Reply:
I'm pretty sure the body you linked is not what is get streamed out from undici. https://github.com/nodejs/undici/blob/master/lib/core/errors.js#L21 as you can see this is an actual error being emitted.